src/authentication/json-api-authorization-spec.ts

```diff
@@ -183,6 +187,49 @@ describe('authorization for api routes', () => {
         expect(api.postProjectHandler).to.not.have.been.called;
       });
     });
+    describe('getDeploymentCommentsHandler', () => {
+      it('should allow fetching comments for a deployment in an authorized project', async () => {
+        // Arrange
+        const { server, authentication, api } = await arrange(
+          'userHasAccessToDeployment',
+          true,
+          'getDeploymentCommentsHandler',
+        );
+        // Act
+        await makeRequest(server, '/comments/deployment/1-1', 'GET');
+
+        // Assert
+        expect(authentication.userHasAccessToDeployment).to.have.been.calledOnce;
```

Comment:
In this and the test below the response is not checked since the request handler has been stubbed out in the 'arrange' part and forced to return 200. In the 'not allowed' cases the response is checked since, in general, the request should not reach the handler at all (and so the response code is not determined by the test).